prototype_source/quantization_in_pytorch_2_0_export_tutorial.rst

```diff
@@ -0,0 +1,341 @@
+==============================================================
+
+Today we have `FX Graph Mode
+Quantization <https://pytorch.org/docs/stable/quantization.html#prototype-fx-graph-mode-quantization>`__
+which uses ``symbolic_trace`` to capture the model into a graph, and then
+perform quantization transformations on top of the captured model. In a
+similar way, for Quantization 2.0 flow, we will now use the PT2 Export
+workflow to capture the model into a graph, and perform quantization
+transformations on top of the ATen dialect graph. This approach is expected to
+have significantly higher model coverage, better programmability, and
+a simplified UX.
+
+Prerequisites:
+-----------------------
+
+-  `Understanding of torchdynamo concepts in PyTorch <https://pytorch.org/docs/stable/dynamo/index.html>`__
+-  `Understanding of the quantization concepts in PyTorch <https://pytorch.org/docs/master/quantization.html#quantization-api-summary>`__
+-  `Understanding of FX graph mode post training static quantization <https://pytorch.org/tutorials/prototype/fx_graph_mode_ptq_static.html>`__
+-  `Understanding of BackendConfig in PyTorch Quantization FX Graph Mode <https://pytorch.org/tutorials/prototype/backend_config_tutorial.html?highlight=backend>`__
+-  `Understanding of QConfigMapping in PyTorch Quantization FX Graph Mode <https://pytorch.org/tutorials/prototype/backend_config_tutorial.html#set-up-qconfigmapping-that-satisfies-the-backend-constraints>`__
+
+Previously in ``FX Graph Mode Quantization`` we were using ``QConfigMapping`` for users to specify how the model to be quantized
+and ``BackendConfig`` to specify the supported ways of quantization in their backend.
+This API covers most use cases relatively well, but the main problem is that this API is not fully extensible
+with two main limitations:
+
+-  Limitation around expressing quantization intentions for complicated operator patterns such as in the discussion of
+   `issue-96288 <https://github.com/pytorch/pytorch/issues/96288>`__ to support ``conv add`` fusion with oneDNN library.
+   It also requires some changes to current already complicated pattern matching code such as in the
+   `PR-97122 <https://github.com/pytorch/pytorch/pull/97122>`__ to support ``conv add`` fusion.
+-  Limitation around supporting user's advanced quantization intention to quantize their model. For example, if backend
+   developer only want to quantize inputs and outputs when the ``linear`` has a third input, it requires co-work from quantization
+   team and backend developer.
+
+To address these scalability issues, 
+`Quantizer <https://github.com/pytorch/pytorch/blob/3e988316b5976df560c51c998303f56a234a6a1f/torch/ao/quantization/_pt2e/quantizer/quantizer.py#L160>`__
+is introduced for quantization in PyTorch 2.0 export. ``Quantizer`` is a class that users can use to
+programmably set the observer or fake quant objects for each node in the model graph. It adds flexibility
```

Comment:
`programmably set the observer or fake quant objects for each node in the model graph.` I think this is out of date, we should say `programmably set the quantization specifications for input and output of each node in the model graph.`

Reply:
Thanks and Fixed.